File: Coordinate Conversions Source/LbcbConversions/src/RotationalMatrix.cs
using MathNet.Numerics.LinearAlgebra.Double;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LbcbConversions
{
    public enum RotationalOrientation
    {
        Roll,
        Pitch,
        Yaw
    }
    public class RotationalMatrix
    {
        private RotationalOrientation orient;
        public RotationalMatrix(RotationalOrientation orient)
        {
            this.orient = orient;
        }
        public DenseMatrix create(double radians)
        {
            double sine = Math.Sin(radians);
            double cosine = Math.Cos(radians);
            DenseMatrix result = DenseMatrix.CreateIdentity(3);
            switch (orient)
            {
                case RotationalOrientation.Roll:
                    result[0, 0] = cosine;
                    result[1, 2] = -sine;
                    result[2, 1] = sine;
                    result[2, 2] = cosine;
                    break;
                case RotationalOrientation.Pitch:
                    result[0, 0] = cosine;
                    result[0, 2] = sine;
                    result[2, 0] = -sine;
                    result[2, 2] = cosine;
                    break;
                case RotationalOrientation.Yaw:
                    result[0, 0] = cosine;
                    result[0, 1] = -sine;
                    result[1, 0] = sine;
                    result[1, 1] = cosine;
                    break;
            }
            return result;
        }
    }
}

File: Coordinate Conversions Source/matlab/rotationalMatrix.m
function rot = rotationalMatrix(angle, type)
rot = eye(3);
switch type
    case 1
        rot(2,2) =  cos( angle );
        rot(2,3) = -sin( angle );
        rot(3,2) =  sin( angle );
        rot(3,3) =  cos( angle );
        return;
    case 2
        rot(1,1) =  cos( angle );
        rot(1,3) =  sin( angle );
        rot(3,1) = -sin( angle );
        rot(3,3) =  cos( angle );
        return;
    case 3
        rot(1,1) =  cos( angle );
        rot(1,2) = -sin( angle );
        rot(2,1) =  sin( angle );
        rot(2,2) =  cos( angle );
end
return;
end
